src/main/java/org/polyjdbc/core/schema/model/Schema.java

```diff
@@ -57,14 +64,31 @@ void add(Sequence sequence) {
     }
 
     public RelationBuilder addRelation(String name) {
-        return RelationBuilder.relation(this, name);
+        if ((schemaName == null) || (schemaName.isEmpty()))
+            return RelationBuilder.relation(this, name);
+        else
+            return RelationBuilder.relation(this, name, getSchemaName());
     }
 
     public IndexBuilder addIndex(String name) {
-        return IndexBuilder.index(this, name);
+        if ((schemaName == null) || (schemaName.isEmpty()))
```

Comment:
this statement is repeated 3 times, you should check input data only once, in a constructor, and replace empty schemaName with null

btw are you sure that this if is necessary? looks like `IndexBuilder.index(this, name, getSchemaName())` is happy with null schemaName